### PR TITLE
Add grpc-gateway

### PR DESCRIPTION
**Pull request by sacloudy**
* Add grpc-gateway for stream data model to generate RESTFul API.
* the modification in banyand/liasion/http/server.go is uncompleted.

### CODE REVIEW

**Comment by hanahmily** [comment]
@remGigo 

All API endpoints are under `/api`. for example, to get a named `trace` stream, one could issue `curl http://localhost:17913/api/v1/stream/schema/trace`.

UI related assets are under `/`

File: api/proto/banyandb/measure/v1/rpc.proto [hunk 1]
```diff
@@ -25,9 +25,15 @@ package banyandb.measure.v1;
 import "banyandb/measure/v1/query.proto";
 import "banyandb/measure/v1/write.proto";
 import "banyandb/measure/v1/topn.proto";
+import "google/api/annotations.proto";
 
 service MeasureService {
-  rpc Query(banyandb.measure.v1.QueryRequest) returns (banyandb.measure.v1.QueryResponse);
+  rpc Query(banyandb.measure.v1.QueryRequest) returns (banyandb.measure.v1.QueryResponse){
```
Comment (hanahmily) [inline api/proto/banyandb/measure/v1/rpc.proto:31]: You should set up `base_path` in this file

File: api/proto/banyandb/stream/v1/rpc.proto [hunk 2]
```diff
@@ -24,8 +24,14 @@ package banyandb.stream.v1;
 
 import "banyandb/stream/v1/query.proto";
 import "banyandb/stream/v1/write.proto";
+import "google/api/annotations.proto";
 
 service StreamService {
```
Comment (hanahmily) [inline api/proto/banyandb/stream/v1/rpc.proto:29]: lack of `base_path`

**Review by hanahmily — CHANGES_REQUESTED** [review]
Some files are lack of `base_path`

**Comment by hanahmily** [comment]
Fixes https://github.com/apache/skywalking/issues/9044

**Review by hanahmily — DISMISSED** [review]

File: ui/.gitignore [hunk 3]
```diff
@@ -18,3 +18,4 @@
 # Binaries for programs and plugins
 dist
 node_modules
+ui-licenses
```
Comment (wu-sheng) [inline ui/.gitignore:21]: What is this folder about?

File: api/proto/banyandb/database/v1/rpc.proto [hunk 4]
```diff
@@ -202,11 +230,33 @@ message MeasureRegistryServiceListResponse {
 }
 
 service MeasureRegistryService {
-  rpc Create(MeasureRegistryServiceCreateRequest) returns (MeasureRegistryServiceCreateResponse);
-  rpc Update(MeasureRegistryServiceUpdateRequest) returns (MeasureRegistryServiceUpdateResponse);
-  rpc Delete(MeasureRegistryServiceDeleteRequest) returns (MeasureRegistryServiceDeleteResponse);
-  rpc Get(MeasureRegistryServiceGetRequest) returns (MeasureRegistryServiceGetResponse);
-  rpc List(MeasureRegistryServiceListRequest) returns (MeasureRegistryServiceListResponse);
+  rpc Create(MeasureRegistryServiceCreateRequest) returns (MeasureRegistryServiceCreateResponse){
```
Comment (lujiajing1126) [inline api/proto/banyandb/database/v1/rpc.proto:233]: CRUD path seems strange. They do not fully follow Rest style

Comment (wu-sheng) [reply api/proto/banyandb/database/v1/rpc.proto:233]: Could you be specific which part?

File: api/proto/banyandb/database/v1/rpc.proto [hunk 5]
```diff
@@ -64,11 +70,33 @@ message StreamRegistryServiceListResponse {
...
-  rpc List(StreamRegistryServiceListRequest) returns (StreamRegistryServiceListResponse);
+  rpc Create(StreamRegistryServiceCreateRequest) returns (StreamRegistryServiceCreateResponse){
+    option(google.api.http) = {
+      post:"/v1/stream/schema"
+      body:"*"
+    };
+  };
+  rpc Update(StreamRegistryServiceUpdateRequest) returns (StreamRegistryServiceUpdateResponse){
+    option(google.api.http) = {
+      put:"/v1/stream/schema"
```
Comment (lujiajing1126) [inline api/proto/banyandb/database/v1/rpc.proto:81]: "/v1/stream/schema/{metadata.group}/{metadata.name}"

File: api/proto/banyandb/database/v1/rpc.proto [hunk 6]
```diff
@@ -64,11 +70,33 @@ message StreamRegistryServiceListResponse {
...
+    };
+  };
+  rpc Get(StreamRegistryServiceGetRequest) returns (StreamRegistryServiceGetResponse){
+    option(google.api.http) = {
+      get:"/v1/stream/schema/{metadata.group}/{metadata.name}"
+    };
+  };
+  rpc List(StreamRegistryServiceListRequest) returns (StreamRegistryServiceListResponse){
+    option(google.api.http) = {
+      get:"/v1/stream/schema-group/{group}"
```
Comment (lujiajing1126) [inline api/proto/banyandb/database/v1/rpc.proto:97]: get:"/v1/stream/schema/{group}"

File: api/proto/banyandb/database/v1/rpc.proto [hunk 7]
```diff
@@ -64,11 +70,33 @@ message StreamRegistryServiceListResponse {
...
 
 service StreamRegistryService {
-  rpc Create(StreamRegistryServiceCreateRequest) returns (StreamRegistryServiceCreateResponse);
-  rpc Update(StreamRegistryServiceUpdateRequest) returns (StreamRegistryServiceUpdateResponse);
-  rpc Delete(StreamRegistryServiceDeleteRequest) returns (StreamRegistryServiceDeleteResponse);
-  rpc Get(StreamRegistryServiceGetRequest) returns (StreamRegistryServiceGetResponse);
-  rpc List(StreamRegistryServiceListRequest) returns (StreamRegistryServiceListResponse);
+  rpc Create(StreamRegistryServiceCreateRequest) returns (StreamRegistryServiceCreateResponse){
+    option(google.api.http) = {
+      post:"/v1/stream/schema"
```
Comment (lujiajing1126) [inline api/proto/banyandb/database/v1/rpc.proto:75]: post:"/v1/stream/schema/{metadata.group}/{metadata.name}"

File: api/proto/banyandb/database/v1/rpc.proto (hunk 4 above)
Comment (lujiajing1126) [reply api/proto/banyandb/database/v1/rpc.proto:233]: > Could you be specific which part?

Comments added. But I am not sure if this is attributed to the limitation of `grpc-gateway`

File: ui/.gitignore (hunk 3 above)
Comment (hanahmily) [reply ui/.gitignore:21]: a temp folder for UI-related license files, the build script will remove it eventually.

File: api/proto/banyandb/database/v1/rpc.proto (hunk 4 above)
Comment (hanahmily) [reply api/proto/banyandb/database/v1/rpc.proto:233]: I will tweak them a bit

File: api/proto/banyandb/database/v1/rpc.proto (hunk 6 above)
Comment (hanahmily) [reply api/proto/banyandb/database/v1/rpc.proto:97]: we have to use `/v1/stream/schema/lists` instead here.

File: api/proto/banyandb/database/v1/rpc.proto (hunk 7 above)
Comment (hanahmily) [reply api/proto/banyandb/database/v1/rpc.proto:75]: restful api doesn't apply path parameters to a `post` endpoint.

File: api/proto/banyandb/database/v1/rpc.proto (hunk 5 above)
Comment (hanahmily) [reply api/proto/banyandb/database/v1/rpc.proto:81]: This patch is good for me.

**Review by hanahmily — APPROVED** [review]

**Pull request merged**